Update gem update protocols (#13962)
chef/chef: CONTRIBUTING.md

------------------------------------------------------------------------
```

### Gems

When making a PR, if you need to add or remove gems, it should be done using the `--conservative` flag:
```shell
gem install [gem_name] --conservative
gem update [gem_name] --conservative
```

> If a manual PR updates the `Gemfile.lock`, please include the output of your `gem update` or `gem install` command in the PR.

Reviewer: This implies there's some difference between "manual" PRs and other PRs. The "manual" from our notes was referring to doing a PR as opposed to doing large-scale gem updates, it was super confusing. So I would change this to something like:

```suggestion
If your PR includes such an update to `Gemfile.lock`, please include the output of your `gem update` or `gem install` command in the PR.
```

Reviewer 2: It was also partly just me mentally distinguishing an update to `Gemfile.lock` vs. a dependbot bump, but "manual" is redundant in that case because it wouldn't be a PR from a human.

------------------------------------------------------------------------
```

> ### Gems
> 
> When installing or updating individual gems, run the install or update commands with the `--conservative` flag.
> 
> ```
> gem install [gem_name] --conservative
> gem update [gem_name] --conservative`
> ```
> 
> If a manual PR updates the `Gemfile.lock`, please include the output of your `gem update` or `gem install` command in the PR.
> 

Reviewer: ```suggestion
When making a PR, if you need to add or remove gems, it should be done using the `--conservative` flag:

```shell
gem install [gem_name] --conservative
gem update [gem_name] --conservative`
```

If a manual PR updates the `Gemfile.lock`, please include the full output of your `gem` command in the commit message an PR description.

```

### Gem Maintenance

At a minimum, once a quarter the maintainers will run a full `bundle update` to update all gems within the limits defined by the `Gemfile`.

> (This should be done preferably **after** minor release for testability)
> 

Reviewer: ```suggestion
(This should be done preferably **after** minor release for testability)

At least once a year, we will also do a review of the limits in the `Gemfile`.
```
